packages/amplify-cli/src/initialize-env.ts

```diff
@@ -42,31 +42,44 @@ export const initializeEnv = async (
     const categoryPluginInfoList = context.amplify.getAllCategoryPluginInfo(context);
     const availableCategories = Object.keys(categoryPluginInfoList).filter(key => initializedCategories.includes(key));
 
-    availableCategories.forEach(category => {
-      categoryPluginInfoList[category].forEach(pluginInfo => {
-        try {
-          // eslint-disable-next-line import/no-dynamic-require, global-require, @typescript-eslint/no-var-requires
-          const { initEnv } = require(pluginInfo.packageLocation);
-
-          if (initEnv) {
-            categoryInitializationTasks.push(() => initEnv(context));
-          }
-        } catch (e) {
-          context.print.warning(`Could not load initEnv for ${category}`);
+    const importCategoryPluginAndQueueInitEnvTask = async (pluginInfo, category) : Promise<void> => {
+      try {
+        const { initEnv } = await import(pluginInfo.packageLocation);
+
+        if (initEnv) {
+          categoryInitializationTasks.push(() => initEnv(context));
```

Comment:
Here Should we throw error in categories initenv task, that will provide a better message and resolution ?

Reply:
actually, here are are not executing the initEnv function.
We are just loading the plugins; and generating an array of Promises that will later execute initEnv down below